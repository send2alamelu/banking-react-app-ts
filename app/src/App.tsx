import React, { Suspense } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Container from '@mui/material/Container';
import Box from '@mui/material/Box';
import CircularProgress from '@mui/material/CircularProgress';
import { ROUTE_PATH } from './constants/RoutePath';
import './App.css';

const LoginPage = React.lazy(() => import('./pages/Login/Login' /* webpackChunkName: "Login" */));
const DashboardPage = React.lazy(() => import('./pages/Dashboard/Dashboard' /* webpackChunkName: "Dashboard" */));
const TransactionPage = React.lazy(() => import('./pages/Transaction/Transaction' /* webpackChunkName: "Transaction" */));

const { LOGIN, DASHBOARD, TRANSACTION } = ROUTE_PATH;

function App() {
  return (
    <React.Fragment>
      <BrowserRouter basename="/banking-react-app-ts">
        <Switch>
          <Suspense fallback={<CircularProgress />}>
            <Container maxWidth="sm">
              <Box sx={{ my: 4 }}>
                <Route exact path={LOGIN} component={LoginPage} />
                <Route path={DASHBOARD} component={DashboardPage} />
                <Route path={TRANSACTION} component={TransactionPage} />
              </Box>
            </Container>
          </Suspense>
        </Switch>
      </BrowserRouter>
    </React.Fragment>
  );
}

export default App;
