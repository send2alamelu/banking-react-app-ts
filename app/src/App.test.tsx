import { render, screen } from '@testing-library/react';
import App from './App';

test('renders banking app', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
