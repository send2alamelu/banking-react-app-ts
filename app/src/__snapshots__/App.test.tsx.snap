// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders banking app 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="content MuiBox-root css-0"
    />
  </div>
</DocumentFragment>
`;
